### PR TITLE
Add new authorization details to README

### PR DESCRIPTION
**Pull request by LeviHassel**
This PR adds more details about Spotify authorization to the README based on the discussion in https://github.com/rinukkusu/spotify-dart/issues/37.

I've personally used every approach I mentioned for Flutter, but I'd appreciate any thoughts from @rinukkusu and @roisnir on how to better explain the Authorization Code Flow for a Dart app.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,30 +17,62 @@ void main() async {
...
+  return SpotifyApi.fromAuthCodeGrant(grant, responseUri);
 }
 ```
 
+Unfortunately, there's not a universal example for implementing the imaginary functions, `redirect` and `listen`, because different options exist for each platform.
+
+For a Flutter app, there's two popular approaches:
+1. Launch a browser using [url_launcher][url_launcher] and listen for a deeplink redirect using [uni_links][uni_links]. The device running the Flutter app will function as the authorization server and handle the redirect. The redirect URI should look something like `com.app://auth`.
+2. Launch a WebView inside the app and listen for a redirect using [webview_flutter][webview_flutter]. This approach requires redirecting to an external authorization server, such as a backend service or serverless function. The redirect URI should be a normal URL, like `https://example.com/auth`, that points to the authorization server.
+
```
Comment (roisnir) [inline README.md:73]: Nice! this clarifies the "imaginary functions" thing.
A third option using webview_flutter as well and listen to the redirect URL with `navigationDelegate`, like this the redirect URL can be made up and do not require backend service to actually listen to it.
```dart
WebView(
  javascriptMode: JavascriptMode.unrestricted,
  initialUrl: authUri,
  navigationDelegate: (navReq) {
    if (!navReq.url.startsWith(redirectUrl))
      return NavigationDecision.navigate;
    final api = SpotifyApi.fromAuthCodeGrant(grant, navReq.url);
    // do somthing with api
    return NavigationDecision.prevent;
  },  ...
```

**Comment by roisnir** [comment]
> This PR adds more details about Spotify authorization to the README based on the discussion in #37.
> 
> I've personally used every approach I mentioned for Flutter, but I'd appreciate any thoughts from @rinukkusu and @roisnir on how to better explain the Authorization Code Flow for a Dart app.

Unfortunately, I haven't written such dart app too, and I have no good idea for production apps but here are some workarounds ideas.
If the case is testing an api request in an example file of a flutter app (like described [here](https://github.com/rinukkusu/spotify-dart/issues/37#issuecomment-608997673)) or any other command-line dart application. You can launch the authorization URL in a web browser ([see this thread](https://stackoverflow.com/questions/33027000/launching-a-web-browser-from-command-line-dart-script)) or print and paste to the browser. Sign in, and then, when redirected copy the redirect URL with the authorization code and paste to the app's input (with stdin.readLineSync())

In the case of an HTTP server, I can't think of a way to authorize because it requires user interface for the user to authenticate.

File: README.md (hunk 1 above)
Comment (LeviHassel) [reply README.md:73]: That's awesome, I'll add that approach to the list!

**Comment by LeviHassel** [comment]
@roisnir thanks for the ideas! I think the command-line approach you described would be a much quicker solution for @hayribakici than what I suggested [here](https://github.com/rinukkusu/spotify-dart/issues/37#issuecomment-609089553).

**Comment by LeviHassel** [comment]
I just updated the PR with code snippets for the `url_launcher`/`uni_links` approach and the WebView approach @roisnir mentioned. I also hid all of the imaginary function explanations inside an expandable/collapsible section so that it doesn't detract from all of the other good advice in the README. 😄

**Comment by rinukkusu** [comment]
Thanks for the great efforts you put into making this documentation better for everyone!

**Pull request merged**